libs/perun/services/src/lib/force-router.service.ts

```diff
@@ -53,4 +71,51 @@ export class ForceRouterService {
   getLastState(): NavigateType {
     return this.lastState;
   }
+
+  /**
+   * This function returns url for application based on appType given in parameter.
+   *
+   * URL of other application differs according to base url of running application
+   * and brands configuration. If base domain doesn't match any url in brands configuration,
+   * then will be used default brand.
+   * If new application url is not defined in brand configuration, then will be used old gui.
+   *
+   * @param appType type of requested app (admin | profile | pwdReset)
+   * @param brands array of brands from configuration
+   * @param login login namespace for pwd reset app
+   */
+  getUrlForOtherApplication(appType: string, brands: Brand[], login?: string): string {
+    const currentUrl = window.location.href;
+    const splittedUrl = currentUrl.split("/");
+    const domain = splittedUrl[0] + "//" + splittedUrl[2]; // protocol with domain
+
+    const brand = ForceRouterService.getBrandContainingDomain(brands, domain);
+    let url: string;
+
+    if (!brand.newApps[appType]) {
+      // url for new app of appType doesn't exist - set url to old gui
+      url = brand.oldGuiDomain;
+      if (this.authService.getIdpFilter()) {
+        url += `/${this.authService.getIdpFilter()}`;
+      }
+      switch (appType) {
+        case "admin":
+          url += "/gui/"
+          break;
+        case "profile":
+          url += "/profile/"
+          break;
+        case "pwdReset":
+          url += "/pwd-reset/"
+          break;
+      }
```

Comment:
this code is not working
dunno what you meant by getIdpFilter(
howewe in old application there needs to be before the name of the application the auth type, so something like /fed/

Reply:
Fixed